com.unity.sg2/Editor/GraphDeltaRegistry/Registry/ContextBuilder.cs

```diff
@@ -1,18 +1,33 @@
 using System;
 using UnityEditor.ShaderFoundry;
+using UnityEditor.ShaderGraph.Defs;
 using static UnityEditor.ShaderGraph.GraphDelta.ContextEntryEnumTags;
 
 namespace UnityEditor.ShaderGraph.GraphDelta
 {
     internal class ContextBuilder : INodeDefinitionBuilder
     {
-        public static readonly RegistryKey kRegistryKey = new RegistryKey { Name = "_Context", Version = 1 };
+        public static readonly RegistryKey kRegistryKey = new RegistryKey {Name = "_Context", Version = 1};
         public RegistryKey GetRegistryKey() => kRegistryKey;
 
         public RegistryFlags GetRegistryFlags() => RegistryFlags.Base;
 
+        public static void AddContextEntry(NodeHandler contextNode, ITypeDescriptor type, string fieldName, Registry registry)
+        {
+            NodeBuilderUtils.ParameterDescriptorToField(
+                new ParameterDescriptor(fieldName, type, GraphType.Usage.In),
+                TYPE.Vec4,
+                contextNode,
+                registry);
```

Comment:
`ParameterDescriptorToField` returns the `PortHandler` that is created. You could pass return that from `AddContextEntry` and pass it directly into the SetupReferableEntry.